src/Compiler/NodeVisitor/AddFormClassesNodeVisitor.php

```diff
@@ -64,15 +70,31 @@ public function enterNode(Node $node): ?Node
             }
             $dimFetch = $assign->expr;
 
+            if (!$this->isUiControl($dimFetch->var)) {
+                return null;
+            }
+
             if (!$dimFetch->dim instanceof String_) {
                 return null;
             }
+
             $formNameString = $dimFetch->dim;
             $formName = $formNameString->value;
 
             $formClassName = $this->formClassNames[$formName] ?? null;
             if ($formClassName === null) {
-                return null;
+                $this->actualForm = new Form($formName, new ObjectType('Nette\Forms\Form'));
+                $error = new Error('Form with name "' . $formName . '" probably does not exist.');
+                return new Assign(
```

Comment:
This is kind of hack :))

Reply:
I took me a lot of thinking but it works :-D